### PR TITLE
[excel] (Samples) Adding guidance on long workbook names

### LINKED ISSUE #560: Error "Office JS error: Line 7: Workbook addWorksheet: A resource with the same name or identifier already exists. clientRequestId: 0e0da3f7-a61b-4618-a919-6b424ead5a19"
[Enter feedback here]

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 688cb5b0-672f-8480-8943-6e4bc403d6b9
* Version Independent ID: 61518540-2796-b66f-4bc7-aef0eca420ad
* Content: [Combine workbooks into a single workbook - Office Scripts](https://learn.microsoft.com/en-us/office/dev/scripts/resources/samples/combine-worksheets-into-single-workbook)
* Content Source: [docs/resources/samples/combine-worksheets-into-single-workbook.md](https://github.com/OfficeDev/office-scripts-docs/blob/main/docs/resources/samples/combine-worksheets-into-single-workbook.md)
* Product: **excel**
* Technology: **scripts**
* GitHub Login: @o365devx
* Microsoft Alias: **o365devx**

### PR DESCRIPTION
**Pull request by AlexJerabek**
Fixes #560.

This PR adds further guidance around the scenario where the combined workbooks have long names.

### CODE REVIEW

File: docs/resources/samples/combine-worksheets-into-single-workbook.md [hunk 1]
```diff
@@ -110,10 +110,17 @@ interface WorksheetData {
...
+- **The argument is invalid or missing or has an incorrect format**: This error can mean that the generated  worksheet name doesn't meet [Excel's requirements](https://support.microsoft.com/office/rename-a-worksheet-3f1f7148-ee83-404d-8ef0-9ff99fbad1f9). This is likely because the name is too long. If the worksheet names will be more than 30 characters replace the code in "Add worksheets" that calls `addWorksheet` with something that shortens the string. Since the workbook name itself might be too long, add an incrementing number to the end of the worksheet name.
 
   ```TypeScript
   let worksheetNumber = 1;
   let worksheetName = `${workbookName}.${value.name}`;
   let sheet = workbook.addWorksheet(`${worksheetName.substr(0,30)}${worksheetNumber++}`);
   ```
+
```
Comment (davidchesnut) [inline docs/resources/samples/combine-worksheets-into-single-workbook.md:120]: In the previous TypeScript, it needs to be broken across the foreach loop as follows (not sure if you want to include this specific or not, but just pointing out this is not a simple "copy paste" operation.

```typescript
function main(workbook: ExcelScript.Workbook, workbookName: string, worksheetInformation: WorksheetData[]) {
  let worksheetNumber = 1;
  
  // Add each new worksheet.
  worksheetInformation.forEach((value) => {
    let worksheetName = `${workbookName}.${value.name}`;
    let sheet = workbook.addWorksheet(`${worksheetName.substr(0, 30)}${worksheetNumber++}`);
```

**Review by davidchesnut — CHANGES_REQUESTED** [review]
Looks good, just a couple notes added.

**Comment by AlexJerabek** [comment]
Thanks @davidchesnut! I've applied your feedback to the PR.

**Comment by AlexJerabek** [comment]
Approved offline. Merging!

**Review by davidchesnut — COMMENTED** [review]
Tested and looks great! (belated approve.)

**Pull request merged**